pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -89,38 +130,117 @@ function make_PyTerminal(app: PyScriptApp) {
+        }
 
-            logger.info('Registering stdio listener');
-            app.registerStdioListener(this);
+        stderr_writeline(msg: string) {
+            this.stdout_writeline(msg);
         }
+        // end of the Stdio interface
+    }
 
-        isAuto() {
-            return this.hasAttribute('auto');
+    return PyTerminalPre;
+}
+
+declare const Terminal: typeof TerminalType;
+
+function make_PyTerminal_xterm(app: PyScriptApp) {
+    /** The <py-terminal> custom element, which automatically register a stdio
+     *  listener to capture and display stdout/stderr
+     */
+    class PyTerminalXterm extends PyTerminalBaseClass {
+        outElem: HTMLDivElement;
+        moduleResolved: boolean;
+        xtermReadyPromise: Promise<TerminalType>;
+        xterm: TerminalType;
+        cachedStdOut: Array<string>;
+        cachedStdErr: Array<string>;
+
+        constructor() {
+            super();
+            this.moduleResolved = false;
+            this.cachedStdOut = [];
+            this.cachedStdErr = [];
+
+            //Required to make xterm appear properly
+            this.style.width = '100%';
+            this.style.height = '100%';
         }
 
-        isDocked() {
-            return this.hasAttribute('docked');
+        async connectedCallback() {
+            this.outElem = document.createElement('div');
+            //this.outElem.className = 'py-terminal';
+            this.appendChild(this.outElem);
+
+            this.setupPosition(app);
+
+            this.xtermReadyPromise = this.setupXterm();
+            await this.xtermReadyPromise;
+        }
+
+        async setupXterm() {
+            // eslint-disable-next-line
+            // @ts-ignore
+            if (globalThis.Terminal == undefined) {
+                //xterm module proper
+                //eslint-disable-next-line
+                //@ts-ignore
+                await import('https://cdn.jsdelivr.net/npm/xterm@5.1.0/lib/xterm.js');
+
+                const cssTag = document.createElement('link');
+                cssTag.type = 'text/css';
+                cssTag.rel = 'stylesheet';
+                cssTag.href = 'https://cdn.jsdelivr.net/npm/xterm@5.1.0/css/xterm.css';
+                document.head.appendChild(cssTag);
+
+                //Create xterm, add addons
+                this.xterm = new Terminal({ screenReaderMode: true, cols: 80 });
+
+                // xterm must only 'open' into a visible DOM element
+                // If terminal is still hidden, open during first write
+                if (!this.autoShowOnNextLine) this.xterm.open(this);
+
+                this.moduleResolved = true;
+
+                //Write out any messages output while xterm was loading
+                this.cachedStdOut.forEach((value: string): void => this.stdout_writeline(value));
+                this.cachedStdErr.forEach((value: string): void => this.stderr_writeline(value));
+            } else {
+                this.moduleResolved = true;
```

Comment:
in which circumstance this is desired/needed/helping? 🤔

Reply:
For situations where the pyterminal object has been constructed, but the xterm.js library has not been loaded from the CDN/set up and therefore cannot have content written to it. With the implementation of `stdout_writeline` and `stderr_writeline` below, this essentially "replays" any writes to stdout that happen before the xterm was ready for input.